### PR TITLE
Keystone v2 and v3 manual and amulet/bundle testing

### PR DESCRIPTION
**Pull request by ajkavanagh**
This provides both keystone v2 and v3 setup for testing barbican
manually.  It demonstrates how to set up a project/tenant in each v3 and
v2 and the how to store a secret.  Then only differences are in the
authentication with Keystone.  However, it does demonstrate both
working.

This change also contains the v2 and v3 keystone tests with barbican.
These are pesented as two gate-_-v2 and gate-_-v3 tests which activate
testing with keystone v2 and keystone v3.

### CODE REVIEW

**Comment by ajkavanagh** [comment]
Still a WIP as just sorting out some final bits of testing.

**Comment by ajkavanagh** [comment]
Ready to go now.

File: src/test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# charm-proof
+charm-tools>=2.0.0
+# amulet deployment helpers
+#bzr+lp:charm-helpers#egg=charmhelpers
+bzr+lp:~ajkavanagh/charm-helpers/add-v3-keystone-catalog-check#egg=charmhelpers
```
Comment (javacruft) [inline src/test-requirements.txt]: This will need updating before we can land this change.

**Comment by gnuoy** [comment]
Please remove the manual_testing directory

**Pull request closed without merging**